modules/common/src/errors.ts

```diff
@@ -95,3 +95,43 @@ export class IpfsFetchError extends Error {
     super("Failed to fetch data from IPFS");
   }
 }
+export class InvalidPermissionError extends Error {
+  constructor() {
+    super("The specified permission does not exist");
+  }
+}
+
+export class InvalidDataError extends Error {
+  constructor() {
+    super("Invalid type, an Uint8Array was expected");
+  }
+}
+
+export class InvalidBigintError extends Error {
+  constructor() {
+    super("Invalid type, BigInt was expected");
+  }
+}
+
+export class InvalidDateError extends Error {
+  constructor() {
+    super("Invalid type, Date was expected");
+  }
+}
+export class InvalidBigNumberishError extends Error {
+  constructor() {
+    super("Invalid type, BigNumberish was expected");
```

Comment:
`BigNumberish` the ethers type?
I don't think we should become ethers-dependent like this

Reply:
We can change this to a big int and close some conversations regarding this